File: app/src/main/java/org/amoustakos/boilerplate/util/ui/ViewUtil.java
package org.amoustakos.boilerplate.util.ui;

import android.app.Activity;
import android.content.Context;
import android.content.res.Resources;
import android.view.inputmethod.InputMethodManager;

import org.amoustakos.boilerplate.injection.ActivityContext;


public final class ViewUtil {

	/**
	 * Translates pixels to dp
	 */
    public static float pxToDp(float px) {
        float densityDpi = Resources.getSystem().getDisplayMetrics().densityDpi;
        return px / (densityDpi / 160f);
    }


	/**
	 * Translates dp to pixels
	 */
	public static int dpToPx(int dp) {
        float density = Resources.getSystem().getDisplayMetrics().density;
        return Math.round(dp * density);
    }


	/**
	 * Hide the keyboard
	 */
	public static void hideKeyboard(@ActivityContext Activity activity) {
        InputMethodManager imm =
                (InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE);
        imm.hideSoftInputFromWindow(activity.getWindow().getDecorView().getWindowToken(), 0);
    }

}

File: utils_android/src/main/java/org/amoustakos/utils/android/ui/ViewUtil.kt
package org.amoustakos.utils.android.ui

import android.app.Activity
import android.content.Context
import android.content.res.Resources
import android.view.inputmethod.InputMethodManager
import android.widget.Toast


object ViewUtil {

    /**
     * Translates pixels to dp
     */
    fun pxToDp(px: Float): Float {
        val densityDpi = Resources.getSystem().displayMetrics.densityDpi.toFloat()
        return px / (densityDpi / 160f)
    }


    /**
     * Translates dp to pixels
     */
    fun dpToPx(dp: Int): Int {
        val density = Resources.getSystem().displayMetrics.density
        return Math.round(dp * density)
    }


    /**
     * Hide the keyboard
     */
    fun hideKeyboard(activity: Activity) {
        val imm = activity.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
        imm.hideSoftInputFromWindow(activity.window.decorView.windowToken, 0)
    }


    fun showToast(context: Context, text: String, isLong: Boolean) {
        Toast.makeText(context, text, if (isLong) Toast.LENGTH_LONG else Toast.LENGTH_SHORT).show()
    }

}
